tools/introspection.c

```diff
@@ -816,6 +843,7 @@ restore_threads(struct ulp_process *process)
   struct ulp_thread *thread;
 
   DEBUG("exiting the critical section (process release).");
+  process->all_threads_hijacked = FALSE;
```

Comment:
OK. I like the fact that you mark the whole process as not hijacked at the very start of the function.